Hardware/Hardware/CardReader/CardReaderEvent.swift

```diff
@@ -1,5 +1,26 @@
+/// An event dispatched by  a connected reader.
+/// It wraps an event type and an optional user-facing message
+public struct CardReaderEvent {
+    public let type: CardReaderEventType
+    public let message: String?
+
+    init(type: CardReaderEventType, message: String? = nil) {
+        self.type = type
+        self.message = message
+    }
+}
+
 /// The possible events from a connected reader.
-public enum CardReaderEvent {
+public enum CardReaderEventType {
+    /// The reader begins waiting for input.
+    /// The app should prompt the customer to present a payment method
+    case waitingForInput
+
+    /// Request that a prompt be displayed in the app.
+    /// For example, if the prompt is SwipeCard,
+    /// the app should instruct the user to present the card again by swiping it.
+    case displayMessage
```

Comment:
Do you think it expresses the intention better to have this enum with the message as an associated value, instead of having an optional message in the `CardReaderEvent` type?

Reply:
I am not sure if it conveys the intention better, but it sure makes things easier for clients, and I don't se any downside to it... I'll update thanks!